docs/examples/writing_drivers/Creating-Instrument-Drivers.ipynb

```diff
@@ -211,46 +223,57 @@
-    "                           label='Voltage',\n",
-    "                           unit='V')\n",
-    "\n",
-    "        self.add_parameter('curr',\n",
-    "                           get_cmd='{}.measure.i()'.format(channel),\n",
-    "                           get_parser=float,\n",
-    "                           set_cmd='{}.source.leveli={}'.format(channel,\n",
-    "                                                                '{:.12f}'),\n",
-    "                           label='Current',\n",
-    "                           unit='A')\n",
-    "\n",
-    "        self.add_parameter('mode',\n",
-    "                           get_cmd='{}.source.func'.format(channel),\n",
-    "                           get_parser=float,\n",
-    "                           set_cmd='{}.source.func={}'.format(channel, '{:d}'),\n",
-    "                           val_mapping={'current': 0, 'voltage': 1},\n",
-    "                           docstring='Selects the output source.')\n",
-    "\n",
-    "        self.add_parameter('output',\n",
-    "                           get_cmd='{}.source.output'.format(channel),\n",
-    "                           get_parser=float,\n",
-    "                           set_cmd='{}.source.output={}'.format(channel,\n",
-    "                                                                '{:d}'),\n",
-    "                           val_mapping={'on':  1, 'off': 0})\n",
-    "\n",
-    "        self.add_parameter('nplc',\n",
-    "                           label='Number of power line cycles',\n",
-    "                           set_cmd='{}.measure.nplc={}'.format(channel,\n",
-    "                                                               '{:.4f}'),\n",
-    "                           get_cmd='{}.measure.nplc'.format(channel),\n",
-    "                           get_parser=float,\n",
-    "                           vals=vals.Numbers(0.001, 25))\n",
+    "        self.volt = Parameter(\n",
+    "            'volt',\n",
+    "            get_cmd='{}.measure.v()'.format(channel),\n",
+    "            get_parser=float,\n",
+    "            set_cmd='{}.source.levelv={}'.format(channel,'{:.12f}'),\n",
+    "            # note that the set_cmd is either the following format string\n",
+    "            #'smua.source.levelv={:.12f}' or 'smub.source.levelv={:.12f}' \n",
+    "            # depending on the value of `channel`\n",
+    "            label='Voltage',\n",
+    "            unit='V',\n",
+    "            instrument=self\n",
+    "        )\n",
+    "\n",
+    "        self.curr = Parameter(\n",
+    "            'curr',\n",
+    "            get_cmd='{}.measure.i()'.format(channel),\n",
+    "            get_parser=float,\n",
+    "            set_cmd='{}.source.leveli={}'.format(channel, '{:.12f}'),\n",
+    "            label='Current',\n",
+    "            unit='A',\n",
+    "            instrument=self\n",
+    "        )\n",
+    "\n",
+    "        self.mode(\n",
```

Comment:
```suggestion
    "        self.mode = Parameter(\n",
```

Reply:
Thanks, applied manually just to be safe